lib/grandpa/grandpa.go

```diff
@@ -535,7 +535,7 @@ func (s *Service) playGrandpaRound() error {
 				return false
 			}
 
-			if completable && finalizable && uint64(s.head.Number.Int64()) >= prevBfc.number {
+			if completable && finalizable && uint32(s.head.Number.Int64()) >= prevBfc.number {
```

Comment:
`uint32(s.head.Number.Int64())` Wouldn't this overflow?

Reply:
block numbers shouldn't go over uint32, in substrate they are set to uint32. but I agree this could be nicer, maybe we can update the `types.Header.Number` type to uint32 at some point